src/plugins/irc-events/connection.js

```diff
@@ -20,6 +21,10 @@ module.exports = function(irc, network) {
 			}), true);
 		}
 
+		if (_.size(client.attachedClients) === 0) {
```

Comment:
Would it make sense to use [`Map`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map) for `client.attachedClients`?
I noticed we use it like this in a couple places, and we could use `client.attachedClients.size` directly (actually, this is one of the arguments of `Object` vs. `Map` used in the MDN article!).